lib/fs.js

```diff
@@ -1836,30 +1836,42 @@ fs.FileWriteStream = fs.WriteStream; // support the legacy name
 
 
 WriteStream.prototype.open = function() {
+  const self = this;
+
   fs.open(this.path, this.flags, this.mode, function(er, fd) {
```

Comment:
According to my [benchmark](https://gist.github.com/brendanashworth/be02fcadce8f441a7a3b), arrow functions actually seem to be reliably faster than regular functions.